### PR TITLE
Remove deprecated `GPUStatsMonitor` callback

### PR DESCRIPTION
**Pull request by rschireman**
## What does this PR do?

Fixes part of #12521 remove `pytorch_lightning.callbacks.gpu_stats.GPUStatsMonitor`

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [ x] Check that all items from **Before submitting** are resolved
- [x ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -63,7 +63,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Removed the deprecated `progress_bar_refresh_rate` argument from the `Trainer` constructor ([#12514](https://github.com/PyTorchLightning/pytorch-lightning/pull/12514))
 
 
--
+- Removed deprecated `pytorch_lightning.callbacks.gpu_stats.GPUStatsMonitor` ([#12554](https://github.com/PyTorchLightning/pytorch-lightning/pull/12554))
```
Comment (rohitgr7) [inline CHANGELOG.md:66]:
```suggestion
- Removed deprecated `GPUStatsMonitor` callback ([#12554](https://github.com/PyTorchLightning/pytorch-lightning/pull/12554))
```

**Review by rohitgr7 — APPROVED** [review]

File: pytorch_lightning/callbacks/__init__.py [hunk 2]
```diff
@@ -15,7 +15,6 @@
 from pytorch_lightning.callbacks.device_stats_monitor import DeviceStatsMonitor
 from pytorch_lightning.callbacks.early_stopping import EarlyStopping
 from pytorch_lightning.callbacks.finetuning import BackboneFinetuning, BaseFinetuning
-from pytorch_lightning.callbacks.gpu_stats_monitor import GPUStatsMonitor
```
Comment (akihironitta) [inline pytorch_lightning/callbacks/__init__.py:18]: Could you also remove it from `__all__` below?

File: tests/deprecated_api/test_remove_1-7.py [hunk 3]
```diff
@@ -379,11 +378,6 @@ def test_v1_7_0_deprecated_slurm_job_id():
 
 
 @RunIf(min_gpus=1)
```
Comment (akihironitta) [inline tests/deprecated_api/test_remove_1-7.py:380]: Please also remove this mark `RunIf` :)
```suggestion
```

**Review by akihironitta — COMMENTED** [review]
Thank you for working on this!

**Review by awaelchli — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]

**Comment by rschireman** [comment]
should I remove the `tests.helpers.runif.RunIf` import in `test_remove_1-7.py` so the pep8 check works in the pre-commit test?

`tests/deprecated_api/test_remove_1-7.py:42: [F401] 'tests.helpers.runif.RunIf' imported but unused`

**Comment by akihironitta** [comment]
> should I remove the `tests.helpers.runif.RunIf` import in `test_remove_1-7.py` so the pep8 check works in the pre-commit test?
> 
> `tests/deprecated_api/test_remove_1-7.py:42: [F401] 'tests.helpers.runif.RunIf' imported but unused`

@rschireman Yes, please remove the import statement there! Thanks for checking :)

**Comment by rschireman** [comment]
@awaelchli  Thanks for removing that import! Now the TPU test is failing -- I looked through the output and couldn't find where it failed, only that it failed on deploying to the Kubernetes cluster. I also noticed that my other pull request (#12658) has this issue. Any thoughts?

**Comment by luigicampanaro** [comment]
Hi! Is there anything implementing the functionality of `GPUStatsMonitor`? I logged `DeviceStatsMonitor` but I got 125 plots (x2), and I'm not sure any of them is relevant to GPU `memory_utilization` and `gpu_utilization`

**Pull request merged**